hack/heap/hh_shared.c

```diff
@@ -148,7 +148,7 @@
  * with the MAP_ANONYMOUS flag. The memfd_create() system call first
  * appeared in Linux 3.17.
  ****************************************************************************/
-#if !defined __APPLE__ && !defined _WIN32
+#ifdef __linux__
```

Comment:
Well those APIs are just posix, so it's fine.  FreeBSD has an extension flag `SHM_ANON` which makes an anonymous map automatically but I don't think it's worth diverging for that.  I'll update the patch to make use of the posix variant.  In general your system detection flags should probably be along the lines of:

```
#ifdef _WIN32
  /* Windows */
#else
  /* Unix-ish (POSIX&Linux) */
  #ifdef __linux__
    /* Linux-isms */
  #else
    /* POSIX */
    #if defined(__APPLE__)
    #elif defined(__FreeBSD__)
    #elif defined(/* other posix variant */)
    #end
  #end
#end
```